### PR TITLE
Test cases

### PR DESCRIPTION
**Pull request by ssingh3006**
- Created unit test cases for users service
- Created unit test cases for App logger
- Created e2e test cases for Request guard

### CODE REVIEW

File: src/modules/users/user.stub.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { User } from './entities/user.entity';
```
Comment (santoshshinde2012) [inline src/modules/users/user.stub.ts:1]: Why need this stub in users module ?

File: test/app.e2e-spec.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { INestApplication } from '@nestjs/common';
 import * as request from 'supertest';
```
Comment (santoshshinde2012) [inline test/app.e2e-spec.ts:3]: Please take a latest from main branch

Comment (santoshshinde2012) [reply test/app.e2e-spec.ts:3]: image

File: test/core.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import * as request from 'supertest';
+import { Test, TestingModule } from '@nestjs/testing';
```
Comment (santoshshinde2012) [inline test/core.e2e-spec.ts:2]: Take a latest , All e2e test cases should be under e2e folder

File: test/logger.spec.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { INestApplication } from '@nestjs/common';
```
Comment (santoshshinde2012) [inline test/logger.spec.ts:1]: All unit test cases should be under *test/unit* folder

File: test/users.response.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+export const users = [{
+    "id": 1,
```
Comment (santoshshinde2012) [inline test/users.response.ts:2]: create *mock* folder to maintain mock data

File: test/users.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
+import { Test, TestingModule } from '@nestjs/testing';
```
Comment (santoshshinde2012) [inline test/users.service.spec.ts:1]: should be under *test/unit*

File: test/users.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  describe('when save is called', () => {
+    let createuserDto: CreateUserDto;
+    beforeEach(async () => {
+      createuserDto = {
+        firstName: userStub().firstName,
+        lastName: userStub().lastName,
+      };
+      await usersService.save(createuserDto);
+    });
+    it('then it should be called with a createUserDTO', () => {
```
Comment (santoshshinde2012) [inline test/users.service.spec.ts:74]: Test cases description should be meaningful

File: test/users.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+      expect(mockUsersRepository.createUser).toHaveBeenCalledWith({
+        firstName: userStub().firstName,
+        lastName: userStub().lastName,
+      });
+    });
+  });
+
+  describe('when update is called', () => {
```
Comment (santoshshinde2012) [inline test/users.service.spec.ts:82]: All the above test cases are related to user service then we don't need create many describe

File: test/users.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+  const mockUsersRepository = {
+    findUser: jest.fn().mockResolvedValue(userStub()),
+    findAllUser: jest.fn().mockResolvedValue([userStub()]),
+    createUser: jest.fn().mockResolvedValue(userStub()),
+    updateUser: jest.fn().mockResolvedValue(userStub()),
+    deleteUser: jest.fn().mockImplementation((dto) => dto),
+  };
+
+  beforeEach(async () => {
```
Comment (santoshshinde2012) [inline test/users.service.spec.ts:21]: Not beforeEach it should be beforeAll

**Review by santoshshinde2012 — COMMENTED** [review]
Please fix the comment but before that take a latest

**Review by santoshshinde2012 — CHANGES_REQUESTED** [review]
Take the latest first

File: src/modules/users/user.stub.ts (hunk 1 above)
Comment (PrajwalKarale08) [reply src/modules/users/user.stub.ts:1]: It was for one sample test cases for checking.

**Pull request closed without merging**